OpenEdXMobile/src/main/java/org/edx/mobile/view/LoginActivity.java

```diff
@@ -235,24 +235,32 @@ public String getEmail() {
         return activityLoginBinding.emailEt.getText().toString().trim();
     }
 
-    private static final int RESET_PASSWORD_REQUEST_CODE = 0;
-
     private void showResetPasswordDialog() {
-        startActivityForResult(ResetPasswordActivity.newIntent(getEmail()), RESET_PASSWORD_REQUEST_CODE);
+        resetPasswordAlert = ResetPasswordAlertDialogFragment.newInstance(this, getEmail());
+        resetPasswordAlert.setListener(this);
```

Comment:
Yeah, I thought about this one a lot.

What I really wanted to do was something like startFragmentForResult(), but it doesn't exist. When I looked up why it didn't exist, the reason was that fragments live inside an activity so communication between activities and their fragments can be more direct.

I also thought about modifying LoginActivity to put the methods right there, then calling it from the fragment (as you suggested). The issue that I had with that is that it didn't seem very modular. By modifying the activity directly, we are intrinsically tying the ResetPasswordDialog with LoginActivity and that didn't feel right to me.

Initially I had the same reservations as you, but then I remembered that if LoginActivity is re-instantiated, the screen would be reset anyway. There isn't any code to make sure that any popup retains its state after being re-instantiated (and if we need to do that we would handle the listener there).

The fragment does have its own lifecycle, but it is dependent on the enclosing activity (in other words, as long as activity is alive, we shouldn't worry about dangling pointers).

So, I think we're safe. But if you want, I can implement your suggestion.